src/chain/chain.rs

```diff
@@ -64,6 +64,8 @@ pub struct Chain {
     event_cache: BTreeSet<NetworkEvent>,
     /// Marker indicating we are processing churn event
     churn_in_progress: bool,
+    /// Marker indicating we are processing a relocation.
+    relocation_in_progress: bool,
```

Comment:
Why do we need another flag instead of using `churn_in_progress` ?

Reply:
Well they mean different thing, so maybe they could be made into an enum, but I would not want to overload the churn_in_progress bool.

churn_in_progress really means, EldersInfo is changeing, while relocation_in_progress EldersInfo will not change

Reply:
As mentionned in PR comment, probably want to clean it up, but maybe not in this PR?